sherpa/stats/__init__.py

```diff
@@ -55,28 +56,40 @@
 
 
 class Stat(NoNewAttributesAfterInit):
-    """The base class for calculating a statistic given data and model."""
+    """The base class for calculating a statistic given data and model.
+
+    .. versionchanged:: 4.17.0
+       Code relying on access to private information may need to be
+       updated, as some fields have changed or been renamed.
```

Comment:
Do we really need a comment in the code to track changes to private fields? That's what "private" implies - we can change it any time; use at your own risk.

Reply:
I think this is just a reflexive change based on the outcome of #2015 - trying to better warn users about possible changes....